common/src/lib.rs

```diff
@@ -2236,13 +2236,28 @@ impl JsonSchema for Node {
 #[cfg_attr(feature = "serde", serde(deny_unknown_fields))]
 #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
 pub struct Tree {
+    /// The identifier of the tree's root node
     pub root: NodeId,
+    /// The name of the application this tree belongs to.
+    /// This value is optional.
```

Comment:
No need to say that it's optional, the type system already communicates this fact.